### DISCUSSION TITLE
basic/stupid question, but need some explaining regarding the BroadcastingEchoServer.cpp code?

### DISCUSSION THREAD

**Discussion by M-U-X** [uNetworking/uWebSockets#1805] [Q&A] [answered]
` for (int i = 0; i < 32; i++) {
                std::string topic = std::to_string((uintptr_t)ws) + "-" + std::to_string(i);
                perSocketData->topics.push_back(topic);
                ws->subscribe(topic);
            }`
            
      
            
wouldn't a single topic suffice? what's the need for 32?

            `.message = [&app](auto *ws, std::string_view message, uWS::OpCode opCode) {
            PerSocketData *perSocketData = (PerSocketData *) ws->getUserData();

            app->publish(perSocketData->topics[(size_t)(++perSocketData->nr % 32)], message, opCode);
            ws->publish(perSocketData->topics[(size_t)(++perSocketData->nr % 32)], message, opCode);
        },`

why are there two publish methods being called? wouldn't the first one the app->publish publish to all clients?
        
I may be misunderstand, but if I'm understanding this correctly this is a server that gets a message from a user then broadcasts to all connected users no?

**Comment by uNetworkingAB** [comment, marked as answer]
This example probably served as a benchmark. So it spams messages like this

**Reply by M-U-X** [reply]
what about the two different publish calls,  the second one is unnecessary? or does it serve a specific purpose?
